### PR TITLE
[OV] Introduce `--quant-mode` cli argument enabling full quantization via optimum-cli

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
# What does this PR do?

- Introduce `--quant-mode` export parameter. Currently the only supported value is `int8` for int8 weights and activations. In the future it can be extended to other combinations, for example `int8/fp8`.
- Introduce `weight_format` and `activation_format` fields to `OVQuantizationConfig`. Currently they are not used as `int8/int8` is the only supported option.
- Enable quantization of speech-to-text pipelines via optimum-cli. Example command is below.

```
optimum-cli export openvino -m openai/whisper-tiny.en --quant-mode int8 --dataset librispeech --num-samples 32 --smooth-quant-alpha 0.9 ./whisper-tiny-en
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/commands/export/openvino.py
Comment (AlexKoff88) [inline optimum/commands/export/openvino.py:81]:
```suggestion
        choices=["int8"],
```
My take is that it can be simplified to just one value if both weights and activations are quantized to the same data type.

Comment (AlexKoff88) [reply optimum/commands/export/openvino.py:81]: Please fix the description above accordingly

File: docs/source/openvino/export.mdx
Comment (AlexKoff88) [inline docs/source/openvino/export.mdx:166]: @echarlaix, @IlyasMoutawwakil, the idea is to expose the capabilities of OVQuantizer in the optimum-cli.

File: optimum/commands/export/openvino.py
Comment (nikita-savelyevv) [reply optimum/commands/export/openvino.py]: Done

**Comment by AlexKoff88** [comment]
@echarlaix, @IlyasMoutawwakil, the PR is ready for your review

**Review by AlexKoff88 — APPROVED** [review]

**Comment by nikita-savelyevv** [comment]
@echarlaix @IlyasMoutawwakil could you please review this PR some time this week? I'm on a vacation starting from the next week. Thanks!

**Comment by AlexKoff88** [comment]
@eaidova, can you please take a look as well.

**Review by echarlaix — APPROVED** [review]
Looks good thanks @nikita-savelyevv, I added a minor comment about an argument name (no worries for us to take care of this part in case you're away at this time, won't block merging)

File: docs/source/openvino/export.mdx
Comment (AlexKoff88) [reply docs/source/openvino/export.mdx:70]: No problem, @echarlaix. The issue with activation-format is that we will let user so much freedom to mix different types of weights and activations data types that we don't support, e.g. W-FP8/A-INT8, W-INT4/A-FP8, etc. We target `--weight-format` for weight-only quantization and it corresponds `nncf.compress_weights()` API while `--quant-mode` controls the type of static full model quantization and corresponds `nncf.quantize(..., mode=)` parameter, e.g. `INT8`, `FP8`, etc.

Comment (AlexKoff88) [reply docs/source/openvino/export.mdx:70]: BTW, we don't assume that `--weight-format` and `--quant-mode` are used all together.

**Comment by AlexKoff88** [comment]
Merging as we have few approvals and no strong objections.

**Pull request merged**
